app/controllers/sign-up.js

```diff
@@ -0,0 +1,45 @@
+import Controller from '@ember/controller';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class SignUpController extends Controller {
+  @tracked email = '';
+
+  @action async createUser(event) {
+    event.preventDefault();
+    const user = this.store.createRecord('user', { email: this.email });
+
+    if (user.isValid) {
+      user.save();
+      this.transitionToRoute('email-verification');
+    }
+  }
```

Comment:
This is probably the biggest "aha" for me in the PR. You make a custom route file if you want to prepare a bunch of things before the user visits the page; you make a custom controller if you want to have an action like a form submission occur in the template/components.

You don't HAVE to create either unless you have something custom to do, because Ember creates them by default.

`this.store.createRecord` returns an instantiated User object, which is based on the User model - for now, it just has an email field, and all the Ember defaults, like `isValid`.

Another of those defaults is a `save` function for the user that tells it to persist the data to the API. That's where the ApplicationAdapter comes in.